editor/editor_inspector.cpp

```diff
@@ -2801,6 +2834,14 @@ void EditorInspector::update_tree() {
 			main_vbox->add_child(category);
 			category_vbox = nullptr; //reset
 
+			// `hint_script` should contain a native class name or a script path.
+			// Otherwise the category was probably added via `@export_category` or `_get_property_list()`.
+			if (p.hint_string.is_empty()) {
+				category->label = p.name;
+				category->set_tooltip_text(p.name);
+				continue; // Do not add an icon, do not change the current class (`doc_name`).
```

Comment:
This was a pretty useful hack, btw, as a way to add icons to categories you could define dummy classes with the same name. But I guess now you can just pass the hint string with `_get_property_list`?

Reply:
Yes, you can still use the hack if you specify a `hint_string`. But property descriptions will be missing or incorrect.

<details>
<summary>Example</summary>

```gdscript
extends Node
@export var a: int
@export_category("Label")
@export var text: int ## Test desc.
```

Before:
![](https://github.com/godotengine/godot/assets/47700418/03ef8914-d79a-4489-aec0-5430f14f7a07)

After:
![](https://github.com/godotengine/godot/assets/47700418/0f22ed45-b771-42c0-aaec-a52122d9f4a2)

</details>

The inspector cannot distinguish between standard properties and added in `_get_property_list()`. But the solution fixes the `@export_category` issue since `hint_string` is empty in this case. We probably shouldn't have added the annotation at all, but people will continue to use it despite the note in the docs.

> Categories in the Inspector dock's list usually divide properties coming from different classes (Node, Node2D, Sprite, etc.). For better clarity, it's recommended to use `@export_group` and `@export_subgroup`, instead.